table/tables/tables.go

```diff
@@ -558,6 +558,38 @@ func (t *Table) AllocAutoID() (int64, error) {
 	return t.alloc.Alloc(t.ID)
 }
 
+// ColumnOffset implements table.Table ColumnOffset interface.
+func (t *Table) ColumnOffset(name string) (int, error) {
+	seps := strings.Split(name, ".")
+	var (
+		table string
+		col   string
+	)
+
+	// Now we only support column or table.column qualified name.
+	// TODO: support db.table.column later.
+	if len(seps) == 1 {
+		col = name
+	} else if len(seps) == 2 {
+		table = seps[0]
+		col = seps[1]
+	} else {
+		return -1, errors.Errorf("invalid column format %s", name)
+	}
+
+	if len(table) > 0 && table != t.Name.O {
```

Comment:
no, alias is only for select fields, but here is for origin table, it can not have alias.